src/array.cr

```diff
@@ -607,20 +641,18 @@ class Array(T)
   # a[6..10]? # => nil
   # a[6..]?   # => nil
   # ```
-  def []?(range : Range)
+  def []?(range : Range) : self | Nil
```

Comment:
in the docs for this method, I think it'd be nice to change the wording from " range's start" to "Range.begin":
```Like `#[Range]` but returns `nil` if the Range.begin is out of range.```